propagation/w3c.go

```diff
@@ -0,0 +1,181 @@
+	if err != nil {
+		return nil, err
+	}
+
+	spanCtx := trace.SpanContext{
+		TraceID:    traceID,
+		SpanID:     spanID,
+		TraceFlags: prop.TraceFlags,
+	}
+
+	return otelSpan{
+		ctx: spanCtx,
+	}, nil
+}
+
+// otelSpan is an implementation of the open telemetry trace.Span interface. We
+// want an implementation in order to use the existing otel code for w3c trace
+// propagation parsing rather than duplicating that code here. That interface is
+// fairly wide, so there are a lot of methods on this type that are noops. The
+// only field we need in order to use their trace context propagator is the
+// trace.SpanContext, so we populate that.
+type otelSpan struct {
+	ctx trace.SpanContext
+}
+
+// SpanContext returns the trace.SpanContext, which is the only field expected to exist.
+func (os otelSpan) SpanContext() trace.SpanContext {
+	return os.ctx
+}
+
+// IsRecording returns false. It exists to satisfy the trace.Span interface.
+func (os otelSpan) IsRecording() bool {
+	return false
+}
+
+// SetStatus does nothing. It exists to satisfy the trace.Span interface.
+func (os otelSpan) SetStatus(code codes.Code, msg string) {
+	return
+}
+
+// SetAttribute does nothing. It exists to satisfy the trace.Span interface.
+func (os otelSpan) SetAttribute(k string, v interface{}) {
+	return
+}
+
+// SetAttributes does nothing. It exists to satisfy the trace.Span interface.
+func (os otelSpan) SetAttributes(attributes ...kv.KeyValue) {
+	return
+}
+
+// End does nothing. It exists to satisfy the trace.Span interface.
+func (os otelSpan) End(options ...trace.EndOption) {
+	return
+}
+
+// RecordError does nothing. It exists to satisfy the trace.Span interface.
+func (os otelSpan) RecordError(ctx context.Context, err error, opts ...trace.ErrorOption) {
+	return
+}
+
+// Tracer returns nil. It exists to satisfy the trace.Span interface.
+func (os otelSpan) Tracer() trace.Tracer {
+	return nil
+}
+
+// AddEvent does nothing. It exists to satisfy the trace.Span interface.
+func (os otelSpan) AddEvent(ctx context.Context, name string, attrs ...kv.KeyValue) {
+	return
+}
+
+// AddEventWithTimestamp does nothing. It exists to satisfy the trace.Span interface.
+func (os otelSpan) AddEventWithTimestamp(ctx context.Context, timestamp time.Time, name string, attrs ...kv.KeyValue) {
+	return
+}
+
+// SetName does nothing. It exists to satisfy the trace.Span interface.
+func (os otelSpan) SetName(name string) {
+	return
+}
+
+// supplier is a container for values, which is a map of strings to strings. It is intended to
+// hold http headers used by the OpenTelemetry SDK. It exists to satisfy the method signatures
+// for the opentelemetry sdk but is not part of the beeline trace API.
+type supplier struct {
+	values map[string]string
+}
+
+// newSupplier creates and returns an empty supplier with an initialized map for values.
+func newSupplier() *supplier {
+	m := &supplier{}
+	m.values = make(map[string]string)
+	return m
+}
+
```

Comment:
If you take the suggestion above about skipping iteration on the passed-in headers, I think this code will be unused and can be removed.
```suggestion
```